### ISSUE TITLE
`ggtags-fontify-code` should be called optionally

### ISSUE THREAD

**Issue by ooglyhLL** [leoliu/ggtags#216]
Some may prefer no highlighting at all. Also, some modes require (significantly) more context for proper highlighting than just a single line[1]. What about a variable to configure the behaviour? Something along these lines:

`ggtags-maybe-fontify-code`:
  `t`: Fontify if the code buffer has `font-lock-mode` switched on.
  `nil`: Force off

[1] I guess loading the entire file into a temporary buffer, highlighting and extracting the string with text properties might be too expensive in case of large files.

**Comment by leoliu** [comment]
The plan originally was for people to use `(add-function (local 'ggtags-get-definition-function) ...)` to override the default.
Alternatively you can defalias `ggtags-fontify-code` to `identity`.

**Comment by ooglyhLL** [comment]
Yeah that would work, thanks. I guess having a switch for the default implementation won't hurt though.

**Comment by mohkale** [comment]
Yeah, I ended up using [org-mode to render code blocks](#212) with ggtags, mostly because it uses temporary buffers which means tree-sitter or other non-font-lock based highlighting would apply, making it consistent with what I see in the actual buffer (for the most part).
